File: frontend/src/lib/domain/item/Category.ts
import type { BaseItem } from "../BaseItem";

export class Category implements BaseItem{
    public static typeName = "category"
    public type: string = Category.typeName;
    public id: string;
    public title: string;

    public static create(id: string, title: string) : Category{
        const category = new Category();
        category.id = id;
        category.title = title;
        return category;
    }
}

File: backend/domain/Category.cs
namespace domain;

public class Category : IEntity
{
    public Guid Id { get; init; }
    public string Name { get; init; }

    public List<Item> Items { get; init; } = new();

    public void AddItem(Item item)
    {
        Items.Add(item);
    }

    public void RemoveItem(Guid itemId)
    {
        var index = Items.FindIndex(_ => _.Id.Equals(itemId));
        if (index < 0) return;
        
        Items.RemoveAt(index);
    }
    
    public static Category Create(string name)
    {
        return new Category()
        {
            Id = Guid.NewGuid(),
            Name = name
        };
    }
}